### PR TITLE
homebrew_cask: reinstall when force is install option

### LINKED ISSUE #3703: hombrew_cask: Install opt not working as expected
### Summary

If I use the play below it will install VSCodium and Firefox-Nightly on the first try as expected, however if the application gets removed, for example moved to trash from the applications folder instead of being uninstalled through brew, then rerunning this play will not reinstall the application. I was expecting `install_options: force` to force the re-installation of the app like `brew install --cask --force vscodium` would. Maybe I am going about this the wrong way. There is another mention of this I found, but it was not confirmed fixed I believe and the issue was somewhat old. 

### Issue Type

Bug Report

### Component Name

homebrew_cask

### Ansible Version

```console (paste below)
$ ansible --version


ansible [core 2.11.6] 
  config file = None
  configured module search path = ['/Users/brandon/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/lib/python3.9/site-packages/ansible
  ansible collection location = /Users/brandon/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible
  python version = 3.9.7 (default, Sep  3 2021, 04:31:11) [Clang 12.0.5 (clang-1205.0.22.9)]
  jinja version = 2.11.2
  libyaml = False
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general

# /opt/homebrew/lib/python3.9/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 3.8.1  

# /Users/brandon/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 4.0.0  
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

macOS: 11.6 (20G165)
Chip: Apple M1

### Steps to Reproduce

```YAML
---
- name: Setup macOS
  hosts: localhost
  connection: local
  vars:
    homebrew_casks:
      - vscodium
      - firefox-nightly
    
- name: Install Homebrew Casks
  when: ansible_distribution == 'MacOSX'
  homebrew_cask:
    name: "{{ homebrew_casks }}"
    state: present
    path: "/opt/homebrew/bin/brew"
    install_options: "force"
```

### Expected Results

I expected the cask to reinstall

### Actual Results

```console (paste below)
...
TASK [Install Homebrew Casks] *****************************************************************
task path: /Users/brandon/rush/local.yml:69
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: brandon
<127.0.0.1> EXEC /bin/sh -c 'echo ~brandon && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/brandon/.ansible/tmp `"&& mkdir "` echo /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571 `" && echo ansible-tmp-1636700262.7389839-41813-224928292110571="` echo /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571 `" ) && sleep 0'
redirecting (type: modules) ansible.builtin.homebrew_cask to community.general.homebrew_cask
Using module file /Users/brandon/.ansible/collections/ansible_collections/community/general/plugins/modules/homebrew_cask.py
<127.0.0.1> PUT /Users/brandon/.ansible/tmp/ansible-local-40239kok6_ek3/tmpcph82c4y TO /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571/AnsiballZ_homebrew_cask.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571/ /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571/AnsiballZ_homebrew_cask.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/opt/homebrew/opt/python@3.9/bin/python3.9 /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571/AnsiballZ_homebrew_cask.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /Users/brandon/.ansible/tmp/ansible-tmp-1636700262.7389839-41813-224928292110571/ > /dev/null 2>&1 && sleep 0'
ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "accept_external_apps": false,
            "greedy": false,
            "install_options": [
                "force"
            ],
            "name": [
                "vscodium",
                "firefox-nightly"
            ],
            "path": "/opt/homebrew/bin/brew",
            "state": "present",
            "sudo_password": null,
            "update_homebrew": false,
            "upgrade_all": false
        }
    },
    "msg": "Changed: 0, Unchanged: 2"
}
...
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by jatorcasso**
##### SUMMARY
Per [suggestion](https://github.com/ansible-collections/community.general/issues/3703#issuecomment-1020909235), fixed bug that prevented force install of homebrew casks and added integration tests.

Fixes #3703 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
homebrew_cask

### CODE REVIEW

**Comment by Akasurde** [comment]
```
fatal: [testhost]: FAILED! => {
01:00     "changed": false, 
01:00     "invocation": {
01:00         "module_args": {
01:00             "accept_external_apps": false, 
01:00             "greedy": false, 
01:00             "install_options": [], 
01:00             "name": [
01:00                 "moonlight"
01:00             ], 
01:00             "path": "/usr/local/bin:/opt/homebrew/bin", 
01:00             "state": "present", 
01:00             "sudo_password": null, 
01:00             "update_homebrew": false, 
01:00             "upgrade_all": false
01:00         }
01:00     }, 
01:00     "msg": "Warning: Unexpected method 'desc' called on Cask moonlight.\nFollow the instructions here:\n  https://github.com/caskroom/homebrew-cask#reporting-bugs\nError: Cask 'moonlight' definition is invalid: invalid 'depends_on macos' value: \":mojave\""
01:00 }

```

**Comment by jatorcasso** [comment]
> ```
> fatal: [testhost]: FAILED! => {
> 01:00     "changed": false, 
> 01:00     "invocation": {
> 01:00         "module_args": {
> 01:00             "accept_external_apps": false, 
> 01:00             "greedy": false, 
> 01:00             "install_options": [], 
> 01:00             "name": [
> 01:00                 "moonlight"
> 01:00             ], 
> 01:00             "path": "/usr/local/bin:/opt/homebrew/bin", 
> 01:00             "state": "present", 
> 01:00             "sudo_password": null, 
> 01:00             "update_homebrew": false, 
> 01:00             "upgrade_all": false
> 01:00         }
> 01:00     }, 
> 01:00     "msg": "Warning: Unexpected method 'desc' called on Cask moonlight.\nFollow the instructions here:\n  https://github.com/caskroom/homebrew-cask#reporting-bugs\nError: Cask 'moonlight' definition is invalid: invalid 'depends_on macos' value: \":mojave\""
> 01:00 }
> ```

@Akasurde I'm not sure how to fix this aside from skipping the OS X tests.. All of the casks are formatted the same way.

File: tests/integration/targets/homebrew_cask/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+cask: bitcoin-core
```
Comment (felixfontein) [inline tests/integration/targets/homebrew_cask/defaults/main.yml:1]: Please pick something else that's less morally/ethically loaded. Thanks.

**Review by felixfontein — CHANGES_REQUESTED** [review]

File: tests/integration/targets/homebrew_cask/defaults/main.yml (hunk 1 above)
Comment (jatorcasso) [reply tests/integration/targets/homebrew_cask/defaults/main.yml]: done

Comment (felixfontein) [reply tests/integration/targets/homebrew_cask/defaults/main.yml]: Thanks!

**Review by felixfontein — APPROVED** [review]
Looks formally good to me. @Akasurde what do you think?

**Review by Akasurde — APPROVED** [review]

**Comment by felixfontein** [comment]
@jatorcasso thanks for fixing this!
@Akasurde thanks for reviewing!

**Comment by Akasurde** [comment]
@jatorcasso Thanks for putting this together. @felixfontein Thanks for review and merge.

**Pull request merged**